runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/SplittableDoFnOperator.java

```diff
@@ -141,21 +212,182 @@ public void outputWindowedValue(
             sideInputReader,
             executorService,
             10000,
-            Duration.standardSeconds(10)));
+            Duration.standardSeconds(10));
+  }
+
+
+  @Override
+  public void processElement(
+      StreamRecord<WindowedValue<KeyedWorkItem<String, KV<InputT, RestrictionT>>>> streamRecord)
+      throws Exception {
+
+    KeyedWorkItem<String, KV<InputT, RestrictionT>> inputElement =
+        streamRecord.getValue().getValue();
+
+    // Initialize state (element and restriction)
+
+    WindowedValue<KV<InputT, RestrictionT>> windowedValue =
```

Comment:
Yes, I'm also not happy about this. The mean reason for copying is so that I can set that last-resort event-time timer that allows me to block shutting down when our upstream operators have shut down.

I'll think about this, maybe I can set the timer outside of the `ProcessFn`, I probably just have to make sure to set the right state namespace on the timer.